apps/mobile-wallet/src/contexts/SendContext.tsx

```diff
@@ -167,44 +161,58 @@ export const SendContextProvider = ({
     async (callbacks: BuildTransactionCallbacks, shouldSweep: boolean) => {
       if (!address || !toAddress) return
 
+      setShouldSweep(shouldSweep)
+
       try {
-        const data = await buildUnsignedTransactions(address, toAddress, assetAmounts, shouldSweep)
-        if (data) setUnsignedTxData(data)
+        if (shouldSweep) {
+          const txParams = getSweepTxParams(address, toAddress)
+          const fees = await fetchSweepTransactionsFees(txParams)
+          setFees(fees)
+        } else {
+          const txParams = getTransferTxParams(address, toAddress, assetAmounts)
+          const fees = await fetchTransferTransactionsFees(txParams)
+          setFees(fees)
+        }
+
         callbacks.onBuildSuccess()
       } catch (e) {
-        const error = (e as unknown as string).toString()
+        const error = (e as unknown as string).toString().toLowerCase()
 
         if (error.includes('consolidating') || error.includes('consolidate')) {
-          setConsolidationRequired(true)
-          dispatch(
-            openModal({
-              name: 'ConsolidationModal',
-              props: {
-                onConsolidate: () => {
-                  authenticateAndSend(onSendSuccessCallback)
-                },
-                fees: unsignedTxData.fees
-              }
-            })
-          )
-          setOnSendSuccessCallback(() => callbacks.onConsolidationSuccess)
-          await buildConsolidationTransactions()
+          try {
+            const txParams = getSweepTxParams(address, address.hash)
+            const fees = await fetchSweepTransactionsFees(txParams)
+
+            dispatch(
+              openModal({
+                name: 'ConsolidationModal',
+                props: { txParams, onSuccess: callbacks.onConsolidationSuccess, fees }
+              })
+            )
+          } catch (error) {
+            showExceptionToast(error, t('Could not build transaction'))
+          }
+        } else if (error.includes('not enough')) {
```

Comment:
I know you're feeling the same, but urg this way of reacting to errors is still a bit scary. Sharing a list of error codes between frontend and backend would be so much safer. Maybe something we could discuss soon with Hongchao / Thomas.